game-app/game-core/src/main/java/games/strategy/engine/posted/game/pbf/NodeBbForumPoster.java

```diff
@@ -109,6 +111,14 @@ public CompletableFuture<String> postTurnSummary(
     }
   }
 
+  private static String appendOptionalSaveGameLink(
```

Comment:
nitpick, methods should be defined after their first usage.
This gets important at scale for consistency and finding where code is within a file.

Reply:
I think this is probably a moot consideration though. 
I'm concerned from an understanding/maintenance perspective by this line here:
```
 appendOptionalSaveGameLink(
                        text, saveGame != null ? uploadSaveGame(client, token, saveGame) : null))),
                        ```
In that line we have control-flow logic, we execute an action to upload a save game, *and* we take the output from that for formatting!! This is a lot to do in just a few lines. Can you work to refactor this a bit so we are better at doing one thing per line? I think that would go a very long way to making this easier to follow. Otherwise, at this juncture, the original code is complex enough that it's very hard to reason about.

Reply:
Hmm what do you think about moving the call to `uploadSaveGame(client, token, saveGame)` inside of `appendOptionalSaveGameLink`?

Reply:
Perhaps could be reasonable. Though, I don't know if we need to go that "drastic", but perhaps could work. At a minimum I was thinking just to use some intermediate variables and just try to align better with the "one-statement, one-action" principle.

eg:
```
if(saveGame == null) {
   appendOptionalSaveGameLink(text, null);
} else {
   String saveGameLink = uploadSaveGame(client,token,saveGame);
   appendOptionalSaveGameLink(text, saveGameLink);

}
```
 I still see a lot of code smells with the above (eg: side-effect'ful call of 'appendOptionalSaveGameLink', would be much nicer if that were functional), but at least the above is a start towards "sanity"  : )